[BUG] ResourceMojoTest.execute_whenCleanWorkDirectoryFails_shouldThrowException fails on Windows CI

## Description

After merging PR #3812 (fix to clean working directory before resource generation), the Windows CI build is consistently failing due to a test that doesn't properly simulate a directory cleaning failure on Windows.

The test `execute_whenCleanWorkDirectoryFails_shouldThrowException` attempts to make the work directory unwritable to trigger an `IOException` during cleanup. However, the Windows-specific approach of using `setReadOnly()` and `setWritable(false)` on a nested file doesn't actually prevent `FileUtils.cleanDirectory()` from succeeding on Windows.

## Steps to Reproduce

1. Run the test suite on Windows:
   ```bash
   ./mvnw test -Dtest=ResourceMojoTest#execute_whenCleanWorkDirectoryFails_shouldThrowException
   ```

2. Observe the test failure

## Expected Behavior

The test should either:
- Successfully simulate a directory cleaning failure on Windows, OR
- Be skipped on Windows if the behavior cannot be reliably simulated

## Actual Behavior

The test fails with:
```
java.lang.AssertionError: 
Expecting code to raise a throwable.
    at org.eclipse.jkube.maven.plugin.mojo.build.ResourceMojoTest.execute_whenCleanWorkDirectoryFails_shouldThrowException(ResourceMojoTest.java:193)
```

This indicates that `FileUtils.cleanDirectory()` succeeds on Windows even when the nested file has `setReadOnly()` applied, so no exception is thrown.

## Root Cause Analysis

The current test logic (lines 183-189 of `ResourceMojoTest.java`):
```java
boolean isWindows = System.getProperty("os.name").toLowerCase().contains("win");
if (!isWindows) {
  assertThat(resourceMojo.workDir.setWritable(false)).isTrue();
} else {
  nestedFile.setReadOnly();
  nestedFile.setWritable(false);
}
```

On Unix systems, making the parent directory non-writable prevents file deletion. On Windows, file permissions work differently - `setReadOnly()` on a file doesn't prevent its parent directory from being cleaned, and Java's file permission methods have limited effectiveness on Windows.

## Suggested Fix

Consider one of these approaches:

1. **Skip the test on Windows** using JUnit's `@DisabledOnOs(OS.WINDOWS)` annotation
2. **Use a different approach for Windows** such as keeping a file handle open to prevent deletion

## Additional Information

- **CI Failure URL:** Visible in PR #3812 checks
- **Related PR:** #3812
- **Related Issue:** #2758
- **Affected File:** `kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java`

Note: A similar test exists in `KubernetesResourceTaskTest.java` for the Gradle plugin which may have the same issue.

--- a/gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java
+++ b/gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java
@@ -212,6 +212,7 @@ public abstract class AbstractJKubeTask extends DefaultTask implements Kubernete
   protected void cleanWorkDirectory() throws IOException {
     final File workDir = kubernetesExtension.getWorkDirectoryOrDefault();
     if (workDir.exists()) {
+      kitLogger.verbose("Cleaning work directory: %s", workDir);
       FileUtils.cleanDirectory(workDir);
     }
   }

--- a/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
+++ b/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
@@ -14,6 +14,7 @@
 package org.eclipse.jkube.gradle.plugin.task;
 
 import java.io.File;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Paths;
@@ -159,19 +160,14 @@ class KubernetesResourceTaskTest {
     FileUtils.forceMkdir(subDir);
     final File nestedFile = new File(subDir, "file.txt");
     FileUtils.write(nestedFile, "content", StandardCharsets.UTF_8);
-    nestedFile.setReadOnly();
-    nestedFile.setWritable(false);
 
-    KubernetesResourceTask resourceTask = new KubernetesResourceTask(KubernetesExtension.class);
-
-    try {
+    try (FileOutputStream lock = new FileOutputStream(nestedFile)) {
+      KubernetesResourceTask resourceTask = new KubernetesResourceTask(KubernetesExtension.class);
       // When & Then
       assertThatThrownBy(resourceTask::runTask)
         .isInstanceOf(IllegalStateException.class)
         .hasMessage("Failed to clean work directory")
         .hasCauseInstanceOf(IOException.class);
-    } finally {
-      nestedFile.setWritable(true);
     }
   }
 

--- a/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
+++ b/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
@@ -27,6 +27,8 @@ import org.apache.commons.io.FileUtils;
 import org.gradle.api.provider.Property;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -123,7 +125,8 @@ class KubernetesResourceTaskTest {
   }
 
   @Test
-  void runTask_whenCleanWorkDirectoryFails_shouldThrowException() throws Exception {
+  @EnabledOnOs({OS.LINUX, OS.MAC})
+  void runTask_whenCleanWorkDirectoryFails_shouldThrowException_onUnix() throws Exception {
     // Given
     final File workDir = taskEnvironment.getRoot().toPath().resolve("build").resolve("jkube-temp").toFile();
     FileUtils.forceMkdir(workDir);
@@ -131,13 +134,7 @@ class KubernetesResourceTaskTest {
     FileUtils.forceMkdir(subDir);
     final File nestedFile = new File(subDir, "file.txt");
     FileUtils.write(nestedFile, "content", StandardCharsets.UTF_8);
-    boolean isWindows = System.getProperty("os.name").toLowerCase().contains("win");
-    if (!isWindows) {
-      assertThat(workDir.setWritable(false)).isTrue();
-    } else {
-      nestedFile.setReadOnly();
-      nestedFile.setWritable(false);
-    }
+    assertThat(workDir.setWritable(false)).isTrue();
 
     KubernetesResourceTask resourceTask = new KubernetesResourceTask(KubernetesExtension.class);
 
@@ -148,11 +145,33 @@ class KubernetesResourceTaskTest {
         .hasMessage("Failed to clean work directory")
         .hasCauseInstanceOf(IOException.class);
     } finally {
-      if (!isWindows) {
-        workDir.setWritable(true);
-      } else {
-        nestedFile.setWritable(true);
-      }
+      workDir.setWritable(true);
+    }
+  }
+
+  @Test
+  @EnabledOnOs(OS.WINDOWS)
+  void runTask_whenCleanWorkDirectoryFails_shouldThrowException_onWindows() throws Exception {
+    // Given
+    final File workDir = taskEnvironment.getRoot().toPath().resolve("build").resolve("jkube-temp").toFile();
+    FileUtils.forceMkdir(workDir);
+    final File subDir = new File(workDir, "nested");
+    FileUtils.forceMkdir(subDir);
+    final File nestedFile = new File(subDir, "file.txt");
+    FileUtils.write(nestedFile, "content", StandardCharsets.UTF_8);
+    nestedFile.setReadOnly();
+    nestedFile.setWritable(false);
+
+    KubernetesResourceTask resourceTask = new KubernetesResourceTask(KubernetesExtension.class);
+
+    try {
+      // When & Then
+      assertThatThrownBy(resourceTask::runTask)
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Failed to clean work directory")
+        .hasCauseInstanceOf(IOException.class);
+    } finally {
+      nestedFile.setWritable(true);
     }
   }
 

--- a/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
+++ b/gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTaskTest.java
@@ -150,6 +150,8 @@ class KubernetesResourceTaskTest {
     } finally {
       if (!isWindows) {
         workDir.setWritable(true);
+      } else {
+        nestedFile.setWritable(true);
       }
     }
   }

--- a/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java
+++ b/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractJKubeMojo.java
@@ -266,6 +266,7 @@ public abstract class AbstractJKubeMojo extends AbstractMojo implements KitLogge
 
     protected void cleanWorkDirectory() throws IOException {
         if (workDir.exists()) {
+            getKitLogger().verbose("Cleaning work directory: %s", workDir);
             FileUtils.cleanDirectory(workDir);
         }
     }

--- a/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
+++ b/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
@@ -24,6 +24,8 @@ import org.apache.maven.settings.Settings;
 import org.eclipse.jkube.kit.common.KitLogger;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnOs;
+import org.junit.jupiter.api.condition.OS;
 import org.junit.jupiter.api.io.TempDir;
 import org.mockito.InOrder;
 
@@ -173,20 +175,15 @@ class ResourceMojoTest {
   }
 
   @Test
-  void execute_whenCleanWorkDirectoryFails_shouldThrowException() throws Exception {
+  @EnabledOnOs({OS.LINUX, OS.MAC})
+  void execute_whenCleanWorkDirectoryFails_shouldThrowException_onUnix() throws Exception {
     // Given
     resourceMojo.workDir.mkdirs();
     final File subDir = new File(resourceMojo.workDir, "nested");
     subDir.mkdirs();
     final File nestedFile = new File(subDir, "file.txt");
     nestedFile.createNewFile();
-    boolean isWindows = System.getProperty("os.name").toLowerCase().contains("win");
-    if (!isWindows) {
-      assertThat(resourceMojo.workDir.setWritable(false)).isTrue();
-    } else {
-      nestedFile.setReadOnly();
-      nestedFile.setWritable(false);
-    }
+    assertThat(resourceMojo.workDir.setWritable(false)).isTrue();
 
     try {
       // When & Then
@@ -195,11 +192,30 @@ class ResourceMojoTest {
         .hasMessage("Failed to clean work directory")
         .hasCauseInstanceOf(java.io.IOException.class);
     } finally {
-      if (!isWindows) {
-        resourceMojo.workDir.setWritable(true);
-      } else {
-        nestedFile.setWritable(true);
-      }
+      resourceMojo.workDir.setWritable(true);
+    }
+  }
+
+  @Test
+  @EnabledOnOs(OS.WINDOWS)
+  void execute_whenCleanWorkDirectoryFails_shouldThrowException_onWindows() throws Exception {
+    // Given
+    resourceMojo.workDir.mkdirs();
+    final File subDir = new File(resourceMojo.workDir, "nested");
+    subDir.mkdirs();
+    final File nestedFile = new File(subDir, "file.txt");
+    nestedFile.createNewFile();
+    nestedFile.setReadOnly();
+    nestedFile.setWritable(false);
+
+    try {
+      // When & Then
+      assertThatThrownBy(resourceMojo::execute)
+        .isInstanceOf(MojoExecutionException.class)
+        .hasMessage("Failed to clean work directory")
+        .hasCauseInstanceOf(java.io.IOException.class);
+    } finally {
+      nestedFile.setWritable(true);
     }
   }
 }

--- a/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
+++ b/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
@@ -30,6 +30,7 @@ import org.junit.jupiter.api.io.TempDir;
 import org.mockito.InOrder;
 
 import java.io.File;
+import java.io.FileOutputStream;
 import java.nio.file.Path;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -204,18 +205,12 @@ class ResourceMojoTest {
     final File subDir = new File(resourceMojo.workDir, "nested");
     subDir.mkdirs();
     final File nestedFile = new File(subDir, "file.txt");
-    nestedFile.createNewFile();
-    nestedFile.setReadOnly();
-    nestedFile.setWritable(false);
-
-    try {
+    try (FileOutputStream lock = new FileOutputStream(nestedFile)) {
       // When & Then
       assertThatThrownBy(resourceMojo::execute)
         .isInstanceOf(MojoExecutionException.class)
         .hasMessage("Failed to clean work directory")
         .hasCauseInstanceOf(java.io.IOException.class);
-    } finally {
-      nestedFile.setWritable(true);
     }
   }
 }

--- a/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
+++ b/kubernetes-maven-plugin/plugin/src/test/java/org/eclipse/jkube/maven/plugin/mojo/build/ResourceMojoTest.java
@@ -197,6 +197,8 @@ class ResourceMojoTest {
     } finally {
       if (!isWindows) {
         resourceMojo.workDir.setWritable(true);
+      } else {
+        nestedFile.setWritable(true);
       }
     }
   }

